src/cpg_flow/inputs.py

```diff
@@ -0,0 +1,268 @@
+"""
+Metamist wrapper to get input sequencing groups.
+"""
+
+import logging
+
+from cpg_utils.config import config_retrieve, update_dict
+
+from cpg_flow.filetypes import CramPath, GvcfPath
+from cpg_flow.targets import Dataset, MultiCohort, PedigreeInfo, SequencingGroup, Sex
+
+from .metamist import (
+    AnalysisType,
+    Assay,
+    MetamistError,
+    get_cohort_sgs,
+    get_metamist,
+    parse_reads,
+)
+from .utils import exists
+
+_multicohort: MultiCohort | None = None
+
+
+def add_sg_to_dataset(dataset: Dataset, sg_data: dict) -> SequencingGroup:
+    """
+    This is moved here just to reduce code duplication
```

Comment:
I think we should improve this message as it was more contextual in a different PR and less helpful in isolation.